packages/samples/headless-react/src/components/search-box/search-box.fn.tsx

```diff
@@ -0,0 +1,47 @@
+import {useEffect, useState, FunctionComponent} from 'react';
+import {
+  buildSearchBox,
+  SearchBox as HeadlessSearchBox,
+  SearchBoxOptions,
+} from '@coveo/headless';
+import {engine} from '../../engine';
+
+interface SearchBoxProps {
+  searchBox: HeadlessSearchBox;
+}
+
+export const SearchBox: FunctionComponent<SearchBoxProps> = (props) => {
+  const {searchBox} = props;
+  const [state, setState] = useState(searchBox.state);
+  const isEnterKey = (e: React.KeyboardEvent<HTMLInputElement>) =>
+    e.key === 'Enter';
+
+  useEffect(() => searchBox.subscribe(() => setState(searchBox.state)), []);
+
+  return (
+    <div>
+      <input
+        value={state.value}
+        onChange={(e) => searchBox.updateText(e.target.value)}
+        onKeyDown={(e) => isEnterKey(e) && searchBox.submit()}
+      />
+      <ul>
+        {state.suggestions.map((suggestion, index) => {
+          const value = suggestion.rawValue;
+          return (
+            <li key={index} onClick={() => searchBox.selectSuggestion(value)}>
```

Comment:
Index is not recommended as a key! 

... sorry I have that burned into me from when I used to use React a lot. :rofl:

Reply:
Sounds good, I will switch to using the raw value 💯